MCprep_addon/spawner/models.py

```diff
@@ -0,0 +1,1350 @@
+				# if 2.8: swap modifier armature and also copy over anim data
+				if mod.type != 'ARMATURE':
+					continue
+				# if not (mod.object in new_objs or mod.object in src_armas):
+				#       continue # all good (from mod arma target perspective)
+				if mod.object not in src_armas and mod.object not in new_armas:
+					continue # unknown source, no way to know how to fix this
+				target_base = util.nameGeneralize(mod.object.name)
+				if target_base not in new_armas_basnames:
+					continue # shouldn't occur, but skip if it does
+
+				new_target = new_armas[new_armas_basnames.index(target_base)]
+				old_target = src_armas[src_armas_basnames.index(target_base)]
+
+				# for blender 2.8, bug is that target is still old_target
+				# but for blender 2.7, the mod target will already have changed
+				if old_target.animation_data:
+					new_target.animation_data_create()
+					new_target.animation_data.action = old_target.animation_data.action
+					conf.log("Updated animation of armature for instance of "+src_coll.name)
+
+				if mod.object in new_objs:
+					continue # was already the new object target for modifier
+				mod.object = new_target
+				conf.log("Updated target of armature for instance of "+src_coll.name)
+
+	def prep_collection(self, context, block, pre_groups):
+		"""Prep the imported collection, ran only if newly imported (not cached)"""
+
+		# Group method first, move append to according layer
+		for ob in util.get_objects_conext(context):
+			util.select_set(ob, False)
+
+		layers = [False]*20
+		if not hasattr(context.scene, "layers"):
+			# TODO: here add all subcollections to an MCprepLib collection.
+			pass
+		elif self.append_layer==0:
+			layers = context.scene.layers
+		else:
+			layers[self.append_layer-1] = True
+		objlist = []
+		group = None
+		for coll in util.collections():
+			if coll in pre_groups:
+				continue
+			for ob in coll.objects:
+				if ob.name in context.scene.objects:
+					#context.scene.objects.unlink(ob)
+					objlist.append(ob)
+			if util.nameGeneralize(coll.name) == util.nameGeneralize(block):
+				group = coll
+
+		if group is None:
+			return None
+
+		if self.prep_materials is True:
+			for ob in objlist:
+				util.select_set(ob, True)
+			try:
+				bpy.ops.mcprep.prep_materials(
+					autoFindMissingTextures=False,
+					improveUiSettings=False,
+					skipUsage=True) # if cycles
+			except:
+				print("Could not prep materials")
+			for ob in util.get_objects_conext(context):
+				util.select_set(ob, False)
+
+		if hasattr(context.scene, "layers"): # 2.7 only
+			for obj in objlist:
+				obj.layers = layers
+		return group
+
+
+class MCPREP_OT_reload_models(bpy.types.Operator):
+	"""Force reload the Model objects and cache."""
+	bl_idname = "mcprep.reload_models"
+	bl_label = "Reload Models"
+
+	@tracking.report_error
+	def execute(self, context):
+		if not context.scene.model_path.lower().endswith('.blend'):
+			self.report({'WARNING'}, "Meshswap file must be a .blend, try resetting")
```

Comment:
Remember to update references here (and any other references in this file) to rename from Meshswap to Entities.